sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala

```diff
@@ -110,9 +114,16 @@ object AkkaGrpcPlugin extends AutoPlugin {
             toGenerator(g, ScalaBinaryVersion(scalaBinaryVersion.value), generatorLogger))
         },
         // configure the proto gen automatically by adding our codegen:
-        // FIXME: actually specifying separate Compile and Test target stub and languages does not work #194
-        PB.targets ++=
-          targetsFor(sourceManaged.value, akkaGrpcCodeGeneratorSettings.value, akkaGrpcGenerators.value),
+        PB.targets := {
+          val configTargets =
+            targetsFor(sourceManaged.value, akkaGrpcCodeGeneratorSettings.value, akkaGrpcGenerators.value)
+          // To play nice with other sbt-protoc plugins that might have injected their own targets, don't simply
+          // override the key. Append instead, after excluding targets added by parent configurations (transitively)
+          // since they already ran and would cause protoc to freeze as their name collides with the own generators
+          // of the config.
+          val parentConfigsGenerators = akkaGrpcGenerators.all(ancestorConfigsFilter(config)).value.flatten
+          PB.targets.value.filterNot(t => parentConfigsGenerators.contains(t.generator)) ++ configTargets
+        },
         PB.protoSources += sourceDirectory.value / "proto",
```

Comment:
I thought about adding the same "exclude parent config entries" behavior to this key for consistency. That would avoid sending the same input to `protoc` potentially several times (and ending up with the same class file (s) shadowing the one (s) compiled earlier), but I realized that would go against #194.